pkg/generate/app/cmd/template.go

```diff
@@ -38,9 +38,11 @@ func TransformTemplate(tpl *templateapi.Template, client client.TemplateConfigsN
 
 	// ensure the template objects are decoded
 	// TODO: in the future, this should be more automatic
-	if errs := runtime.DecodeList(result.Objects, kapi.Codecs.UniversalDecoder()); len(errs) > 0 {
-		err = errors.NewAggregate(errs)
-		return nil, fmt.Errorf("error processing template %s: %v", name, err)
+	if decode {
+		if errs := runtime.DecodeList(result.Objects, kapi.Codecs.UniversalDecoder()); len(errs) > 0 {
```

Comment:
Huh, looks like `runtime.DecodeList` really is innocent here. Will update PR once I figure out what's wrong.